### PR TITLE
[ios] Added progress to maps download from routing.

### PR DESCRIPTION
**Pull request by igrechuhin**
MAPSME-763

### CODE REVIEW

File: iphone/Maps/Classes/CustomAlert/DownloadTransitMapsAlert/MWMDownloadTransitMapAlert.mm [hunk 1]
```diff
@@ -111,119 +85,136 @@ + (instancetype)downloaderAlertWithMaps:(storage::TCountriesVec const &)maps
...
+  alert.cancelBlock = cancelBlock;
+  alert.downloadBlock = downloadBlock;
+  alert.downloadCompleteBlock = downloadCompleteBlock;
   return alert;
 }
 
-+ (instancetype)alertWithMaps:(storage::TCountriesVec const &)maps
++ (instancetype)alertWithCountries:(storage::TCountriesVec const &)countries
 {
+  NSAssert(!countries.empty(), @"TCountriesVec can not be empty.");
```
Comment (VladiMihaylenko) [inline iphone/Maps/Classes/CustomAlert/DownloadTransitMapsAlert/MWMDownloadTransitMapAlert.mm]: countries can not be empty

File: iphone/Maps/Classes/CustomAlert/DownloadTransitMapsAlert/MWMDownloadTransitMapAlert.mm [hunk 2]
```diff
@@ -111,119 +85,136 @@ + (instancetype)downloaderAlertWithMaps:(storage::TCountriesVec const &)maps
...
-    case SelectionStateBoth:
+    [self.dialogsTableView beginUpdates];
+    [self.dialogsTableView endUpdates];
+  };
+  if (listExpanded)
+  {
+    CGFloat const actualHeight = kCellHeight * m_countries.size() + kHeaderHeight;
+    CGFloat const height = [self bounded:actualHeight withHeight:self.superview.height];
+    self.tableViewHeight.constant = height;
+    self.dialogsTableView.scrollEnabled = actualHeight > self.tableViewHeight.constant ? YES : NO;
```
Comment (VladiMihaylenko) [inline iphone/Maps/Classes/CustomAlert/DownloadTransitMapsAlert/MWMDownloadTransitMapAlert.mm]:
```objc
self.dialogsTableView.scrollEnabled = actualHeight > self.tableViewHeight.constant;
```

File: iphone/Maps/Classes/CustomAlert/DownloadTransitMapsAlert/MWMDownloadTransitMapAlert.mm [hunk 3]
```diff
@@ -111,119 +85,136 @@ + (instancetype)downloaderAlertWithMaps:(storage::TCountriesVec const &)maps
...
 
-- (IBAction)downloadButtonTap:(id)sender
+#pragma mark - MWMFrameworkStorageObserver
+
+- (void)processCountryEvent:(TCountryId const &)countryId
 {
-  [Statistics logEvent:kStatisticsEvent withParameters:@{kStatAction : kStatApply}];
-  [self downloadMaps];
+  auto const overallProgress = GetFramework().Storage().GetOverallProgress(m_countries);
+  if (overallProgress.first == overallProgress.second)
```
Comment (syershov) [inline iphone/Maps/Classes/CustomAlert/DownloadTransitMapsAlert/MWMDownloadTransitMapAlert.mm]: // Test if downloading has finished by comparing downloaded and total sizes.

File: iphone/Maps/Classes/CustomAlert/DownloadTransitMapsAlert/MWMDownloadTransitMapAlert.mm [hunk 4]
```diff
@@ -111,119 +85,137 @@ + (instancetype)downloaderAlertWithMaps:(storage::TCountriesVec const &)maps
...
 
-+ (instancetype)alertWithMaps:(storage::TCountriesVec const &)maps
++ (instancetype)alertWithCountries:(storage::TCountriesVec const &)countries
 {
+  NSAssert(!countries.empty(), @"countries can not be empty.");
   MWMDownloadTransitMapAlert * alert = [[[NSBundle mainBundle] loadNibNamed:kDownloadTransitMapAlertNibName owner:nil options:nil] firstObject];
-  NSMutableArray * missedFiles = [@[] mutableCopy];
-  if (!maps.empty())
+
+  NSMutableArray * titles = [@[] mutableCopy];
```
Comment (VladiMihaylenko) [inline iphone/Maps/Classes/CustomAlert/DownloadTransitMapsAlert/MWMDownloadTransitMapAlert.mm]: NSMutableArray<NSString *> *

**Comment by bykoianko** [comment]
LGTM в плане использования ядра

**Comment by VladiMihaylenko** [comment]
LGTM

**Comment by syershov** [comment]
LGTM

**Pull request merged**